server/chatroom.py

```diff
@@ -20,36 +23,188 @@ def get_msg_update(cur, chatroom_id, date_time):
 
 
 # Retrieve message functionality
-def retrieveMessages(cur, chatroom_id):
-    cur.execute("SELECT * FROM messages WHERE chatroom_id= %(chatroom_id)s", {"chatroom_id": chatroom_id})
-    result = json.dumps(cur.fetchall())
-    return result
+def retrieve_messages(cur, chatroom_id):
+    try:
+        cur.execute("SELECT (sender_id, email, fname, lname, chatroom_id, message_text, cast(date_time_sent as text)) FROM message JOIN student ON student_id = sender_id WHERE chatroom_id= %(chatroom_id)s",
+                     {"chatroom_id": chatroom_id})
+        result = json.dumps(cur.fetchall())
+        return result
+    except BaseException as e:
+        print(f'Exception: {e}')
+        raise HTTPException(
+            status_code=500,
+            detail="Database Error",
+        )
 
 
 # Save message functionality
-def saveMessage(cur, sender_id, chatroomID, message_sent):
+def save_message(cur, sender_id, chatroomID, message_sent):
     try:
         date_time_sent = datetime.datetime.now()
         cur.execute(
             "INSERT INTO message(sender_id, chatroom_id, message_text, date_time_sent) VALUES(%(sender_id)s, %(chatroomID)s, %(message_sent)s, %(date_time_sent)s)",
             {"sender_id": sender_id, "chatroomID": chatroomID, "message_sent": message_sent,
              "date_time_sent": date_time_sent})
         return True
-    except:
-        return False
+    except BaseException as e:
+        print(f'Exception: {e}')
+        raise HTTPException(
+            status_code=500,
+            detail="Database Error",
+        )
 
 
 # Create non-course chatroom
-def createChatroom(cur, user_id, chatroom_name, uni_id):
+def create_chatroom(cur, user_id, chatroom_name, uni_id):
     try:
-        cur.execute("INSERT INTO chatroom(chatroom_name, uni_id) VALUES(%(chatroom_name)s, %(uni_id)s)",
-                    {"chatroom_name": chatroom_name, "uni_id": uni_id})
-        cur.execute("SELECT id FROM chatroom where chatroom_name = %(chatroom_name)s   AND uni_id = %(uni_id)s",
-                    {"chatroom_name": chatroom_name, "uni_id": uni_id})
+        # Generate invite id
+        invite_id = ''.join(random.choices(string.ascii_uppercase + string.digits, k=10))
+        
+        cur.execute("INSERT INTO chatroom(chatroom_name, uni_id, invite_id) VALUES(%(chatroom_name)s, %(uni_id)s, %(invite_id)s)", {"chatroom_name" : chatroom_name, "uni_id" : uni_id, "invite_id" : invite_id})
+        
+        cur.execute("SELECT id FROM chatroom where chatroom_name = %(chatroom_name)s   AND uni_id = %(uni_id)s", {"chatroom_name" : chatroom_name, "uni_id" : uni_id})
```

Comment:
this isnt safe exactly, what if we have 2 chatrooms sharing the same name? ig u should do a count(*) query first on chatroom to get the id we are supposed to do the insert. but if theres a better way to get it, please feel free